app/api/api_v1/endpoints/assignment_router.py

```diff
@@ -28,5 +37,32 @@ async def get_assignments(
         student_assignment = await StudentAssignmentService(db, student, assignment).get_student_assignment_schema()
         student_assignments.append(student_assignment)
         
+    return student_assignments
+
+@router.patch("/assignment/{assignment_name}", response_model=AssignmentSchema)
```

Comment:
why are we using `assignment_name` instead of `assignment_id` here? once the assignment is created in the db it should return an `id` that we can use everywhere, rather than a `name` that potentially has spaces and has to be encoded or whatever, imo. it will prooobably work, but working with IDs is way better than working with long strings if you can help it. what do you think @frostyfan109?

Reply:
we auto populate the assignment_id when creating the assignment. so unless the prof saves the ids that  we return, or unless we create a view where they can see the assignment ids, they won't know what to input for the script no?

Reply:
Oh, I was thinking there would be an endpoint that would input the assignment name and return the assignment id to make sure that the assignment exists first, but I suppose that that's more expensive than just doing things the way you're doing it. Let's keep it this way.